WPMS.c

```diff
@@ -0,0 +1,661 @@
+            printf("User ID: %d\n",booking.user_id);
+            printf("User Name: %s\n",booking.userName);
+            printf("Venue Name: %s\n",booking.venueName);
+            printf("Venue Location: %s\n",booking.location);
+            printf("Booked Date: %d / %d / %d\n",booking.userDate,booking.userMonth,booking.userYear);
+        }
+    }
+    fclose(readBookedDetails);
+}
+void bookAnEvent()
+{
+    int choice,bookedVenue,count=1,bookedMandap,bookedStage,answer,bid;
+    FILE *writeBookingDetails,*readBookingDetails;
+    FILE *readUserDetails,*readVenueDetails;
+    do
+    {
+        system("cls");
+        printf("1. List of Venues\n2. Mandap Decorations\n3. Stage Decorations\n4. Catering Services\n5. Go Back\n");
+        do
+        {
+            printf("Select your choice by entering a number from above options: ");
+            scanf("%d",&choice);
+        }while(choice<1 && choice>5);
+        switch(choice)
+        {
+            case 1: bookedVenue = list_of_venues();
+                    calendar(bookedVenue);
+                    readBookingDetails=fopen("VenuesBooked.txt","rb");
+                    if(readBookingDetails==NULL)
+                        printf("Not there\n");
+                    while(fread(&booking,sizeof(booking),1,readBookingDetails)==1);
+                    bid=booking.booking_id;
+                    booking.booking_id=bid+1;
+                    fclose(readBookingDetails);
+                    readUserDetails=fopen("UserDetails.txt","rb");
+                    while(fread(&user,sizeof(user),1,readUserDetails)==1)
+                    {
+                        if(uid == user.user_id)
+                        {
+                            booking.user_id=user.user_id;
+                            strcpy(booking.userName,user.userName);
+                            break;
+                        }
+                    }
+                    fclose(readUserDetails);
+                    readVenueDetails=fopen("VenueDetails.txt","rb");
+                    while(fread(&venue,sizeof(venue),1,readVenueDetails)==1)
+                    {
+                        if(count==bookedVenue)
+                        {
+                            strcpy(booking.venueName,venue.venueName);
+                            strcpy(booking.location,venue.location);
+                            booking.userDate=cal.userDate;
+                            booking.userMonth=cal.userMonth;
+                            booking.userYear=cal.userYear;
+                            writeBookingDetails=fopen("VenuesBooked.txt","ab");
+                            fwrite(&booking,sizeof(booking),1,writeBookingDetails);
+                            break;
+                        }
+                        count++;
+                    }
+                    fclose(readVenueDetails);
+                    fclose(writeBookingDetails);
+
+                    break;
+            case 2: mandap_decoration(1);

```

Comment:
For type of user ,instead of hardcoding 1 value in function call you can use #define NORMALUSER 1
#define ADMIN 2
such kind of constants and use those constants in the code such as mandap_decoration(NORMALUSER)